src/Bridge/Symfony/Routing/ApiLoader.php

```diff
@@ -55,14 +59,18 @@ public function load($data, $type = null)
 
         foreach ($this->resourceNameCollectionFactory->create() as $resourceClass) {
             $resourceMetadata = $this->resourceMetadataFactory->create($resourceClass);
-            $normalizedShortName = Inflector::pluralize(Inflector::tableize($resourceMetadata->getShortName()));
+            $resourceShortName = $resourceMetadata->getShortName();
+
+            if (null === $resourceShortName) {
+                throw new InvalidResourceException(sprintf('Resource "%s" has no short name defined.', $resourceClass));
```

Comment:
@soyuka @polc We should not expect the `ShortNameResourceMetadataFactory` to always be used (it's just a standard implementation). The check is necessary.